src/ripple/app/main/Application.cpp

```diff
@@ -1034,6 +1034,41 @@ class ApplicationImp
                     << "Remaining free disk space is less than 512MB";
                 signalStop ();
             }
+
+            static std::atomic<std::uint32_t> pageSize {};
+            static std::atomic<std::uint32_t> maxPages;
```

Comment:
Using lambdas to initialize these statics would simplify things. It would get rid of the `std::atomic` and we could make it `const`. Something like:
```
            auto db = mTxnDB->checkoutDb();
            static std::uint32_t const pageSize = [&]{
                std::uint32_t ps;
                *db << "PRAGMA page_size;", soci::into(ps);
                return ps;
            }();
            static std::uint32_t const maxPages = [&]{
                std::uint32_t mp;
                *db << "PRAGMA max_page_count;" , soci::into(mp);
                return mp;
            }();
```

Reply:
Fixed. Good idea to simplify things. Also glad that C++11 lets us do this safely.

Reply:
Since the parameter list is empty, you can use ```static auto const pageSize =```

Reply:
fixed